stac_geoparquet/stac_geoparquet.py

```diff
@@ -90,9 +133,42 @@ def to_geodataframe(
         "expires",  # timestamps extension
         "published",
         "unpublished",
-    ]:
-        if column in gdf.columns:
-            gdf[column] = pd.to_datetime(gdf[column], format="ISO8601")
+    }
+
+    items2["geometry"] = geopandas.array.from_shapely(items2["geometry"])
+
+    if dtype_backend is None:
+        msg = (
+            "The default argument for 'dtype_backend' will change from "
+            "'numpy_nullable' to 'pyarrow'. To keep the previous default "
+            "specify ``dtype_backend='numpy_nullable'``. To accept the future "
+            "behavior specify ``dtype_backend='pyarrow'."
+        )
+        warnings.warn(FutureWarning(msg))
+        dtype_backend = "numpy_nullable"
+
+    if dtype_backend == "pyarrow":
+        for k, v in items2.items():
+            if k in DATETIME_COLUMNS:
+                items2[k] = pd.arrays.ArrowExtensionArray(
+                    pa.array(pd.to_datetime(v, format="ISO8601"))
```

Comment:
I want the output here to be identical to what we're getting in https://github.com/stac-utils/stac-geoparquet/pull/27/.

Right now, the date time columns from this PR end up with nanosecond precision, while Kyle's PR has microsecond precision. I'm not sure if there's a correct default, but we should try and get them the same.